src/storage/mvcc/txn.rs

```diff
@@ -221,26 +221,16 @@ impl<S: Snapshot> MvccTxn<S> {
     // It's possible that lock conflict occours on them, but the isolation is
     // guaranteed by pessimistic locks on row keys, so let TiDB resolves these
     // locks immediately.
-    fn handle_non_pessimistic_lock_conflict(
-        &self,
-        key: Key,
-        for_update_ts: TimeStamp,
-        lock: Lock,
-    ) -> Result<()> {
+    fn handle_non_pessimistic_lock_conflict(&self, key: Key, lock: Lock) -> Result<()> {
         // The previous pessimistic transaction has been committed or aborted.
         // Resolve it immediately.
         //
         // Because the row key is locked, the optimistic transaction will
```

Comment:
If the corresponding pessimistic lock is rolled back because it's timed out and not heartbeating, here we may abort a normally running transaction. Maybe we'd better also improve this case.

Reply:
It's hard to improve.